core/src/com/unciv/logic/map/mapunit/MapUnit.kt

```diff
@@ -802,14 +802,15 @@ class MapUnit : IsPartOfGameInfoSerialization {
         // addPromotion requires currentTile to be valid because it accesses ruleset through it.
         // getAncientRuinBonus, if it places a new unit, does too
         currentTile = tile
+        val improvement = tile.improvement
 
         if (civ.isMajorCiv()
-            && tile.improvement != null
-            && tile.getTileImprovement()!!.isAncientRuinsEquivalent()
+            && improvement != null
+            && tile.ruleset.tileImprovements[improvement]!!.isAncientRuinsEquivalent()
```

Comment:
YES! Non trivial edge case which deserves a comment

Reply:
Would be the wrong place currently to put the comment as it's not relevant at this line yet and it was only changed here for consistency. It would be better to put the comment on the val itself